openassessment/workflow/models.py

```diff
@@ -346,12 +355,61 @@ def _get_steps(self):
             # If no steps exist for this AssessmentWorkflow, assume
             # peer -> self for backwards compatibility
             self.steps.add(
-                AssessmentWorkflowStep(name=self.STATUS.peer, order_num=0),
-                AssessmentWorkflowStep(name=self.STATUS.self, order_num=1)
+                AssessmentWorkflowStep(name=self.STATUS.peer, order_num=1),
+                AssessmentWorkflowStep(name=self.STATUS.self, order_num=2)
             )
             steps = list(self.steps.all())
+
+        # A staff step will always be available, to allow for staff overrides
```

Comment:
I'm 100% on board for adding an exception if needed. Will this "setting of the workflow steps" we're discussing here be critical to that, or will we still be able to check the problem definition separately? I guess in my head it was always:

* Does the problem require a staff grade? Check the problem definition
* Does a staff grade exist? Assume a staff grading step is in place, and check if it has a grade

I could see where this could cause problems if `self.steps` is treated as the canonical source of problem requirements, that just wasn't my understanding.